### ISSUE TITLE
[PLW1514] `write_text` not detected correctly

### ISSUE THREAD

**Issue by mrexodia** [astral-sh/ruff#23278]
### Summary

```py
import json
from pathlib import Path

def get_ida_config_path() -> Path:
    idausr = "."

    return Path(idausr) / "ida-config.json"

def set_ida_config(config: dict):
    ida_config_path = get_ida_config_path()
    if not ida_config_path.exists():
        ida_config_path.parent.mkdir(parents=True, exist_ok=True)

    _ = ida_config_path.write_text(json.dumps(config))

def main():
    with open("hello.txt", "w") as f:
        f.write("Hello from main.py!")
    print("Hello from ruff-encoding-bug!")


if __name__ == "__main__":
    main()
```

```
C:\CodeBlocks\ruff-encoding-bug>uvx ruff check --preview --select PLW1514
PLW1514 `open` in text mode without explicit `encoding` argument
  --> main.py:17:10
   |
16 | def main():
17 |     with open("hello.txt", "w") as f:
   |          ^^^^
18 |         f.write("Hello from main.py!")
19 |     print("Hello from ruff-encoding-bug!")
   |
help: Add explicit `encoding` argument

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

Note that `ida_config_path.write_text(json.dumps(config))` does not have an `encoding` parameter and should also trigger this rule.

### Version

ruff 0.15.1 (a2f11d239 2026-02-12)

**Comment by getehen** [comment]
This seems to be a problem with inferring the type of `Path() / ""` objects (see [this playground](https://play.ruff.rs/f926aa60-c62a-44de-b385-418056abc21a).) Maybe somebody is already working on this issue ? If not I will take a look

**Comment by ntBre** [comment]
I don't _think_ we should be inferring the type of `Path() / ""` at all. Our limited type inference usually relies primarily on type annotations, so I would expect the `get_ida_config_path() -> Path` type to be enough here. Feel free to take a look, though! The rule definitely seems like it's supposed to apply to `Path.write_text`:

https://github.com/astral-sh/ruff/blob/a04d67ec1c53488a2dbd18d66ee8a142106a0faa/crates/ruff_linter/src/rules/pylint/rules/unspecified_encoding.rs#L281

So this seems like a bug.

**Comment by getehen** [comment]
Right, I am making a fix
